pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,431 @@
+        self._data[name] = cim_object.copy()
+
+    def delete(self, name):
+        """
+        Overrides the abstract method.
+
+        If the object exists in the object store dictionary, it is deleted.
+        If it does not exist an exception is returned.
+        """
+        if name in self._data:
+            del self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def iter_names(self):
+        """
+        Overrides the abstract method.
+
+        Returns iterator for names of the objects in the object store
+
+        """
+        return six.iterkeys(self._data)
+
+    def iter_values(self, copy=True):
+        """
+        Overrides the abstract method.
+
+        Returns an iterator for the object values in the object store
+        """
+        for value in six.itervalues(self._data):
+            if copy:
+                yield(value.copy())
+            else:
+                yield(value)
+
+    def len(self):
+        """
+        Overrides the abstract method.
+
+        :class:`~pywbem_mock._baserepository.ObjectStoreAPI.len`
+
+        Returns count of number of objects in the object store dictionary.
+        """
+        return len(self._data)
+
+
+class InMemoryRepository(BaseRepositoryAPI):
+    """
+    Defines the data store and access methods for a simple CIMClass,
+    CIMInstance, and CIMQualifierDeclaration repository that maintains the date
+    in memory.
+
+    The API for this data store is defined in the BaseRepositoryAPI class and
+    the ObjectStoreAPI class.
+    """
+    def __init__(self, initial_namespace=None):
+        """
+        Initialize the InMemoryRepository by creating the top level dictionary.
+        The initial_namespace is optional.  If provided, that namespace
+        is added to the empty repository.
+
+        However, if there is a persistence mechanism, that would define
+        the namespaces from the persistent data so the initial namespace would
+        not be used.
+
+          Parameters:
+             initial_namespace:(:term:`string` or None):
+                Optional initial namespace to be defined for this repository.
+        """
+
+        # Create the in memory repository.  This defines the top level
+        # NocaseDict which defines the namespaces.
+        self._repository = NocaseDict()
+
+        # If an initial namespace is defined, add it to the repository
+        if initial_namespace:
+            self.add_namespace(initial_namespace)
+
+    def print_repository(self, dest=None, ):
+        """
+        Display the items in the repository. This displays information on
+        the items in the data base and is only a diagnostic tool.
+
+        Parameters:
+          dest (:term:`string`):
+            File path of the output file. If `None`, the output is written to
+            stdout.
```

Comment:
That prevents writing to other streams such as strings or stderr. I suggest to change `dest` to become a file-like object.

Reply:
Interesting in that this is a copy of the doc for display repository in _wbemconnection.py. Changed. DONE